lib/rex/proto/http/client.rb

```diff
@@ -20,8 +20,10 @@ class Client
 
   #
   # Creates a new client instance
+  # @param http_trace_proc_request [Proc] A proc object passed to log HTTP requests if HTTP-Trace is set
+  # @param http_trace_proc_response [Proc] A proc object passed to log HTTP responses if HTTP-Trace is set
   #
-  def initialize(host, port = 80, context = {}, ssl = nil, ssl_version = nil, proxies = nil, username = '', password = '', comm: nil)
+  def initialize(host, port = 80, context = {}, ssl = nil, ssl_version = nil, proxies = nil, username = '', password = '', comm: nil, http_trace_proc_request: nil, http_trace_proc_response: nil)
```

Comment:
If this took an object, instead of separate methods, we wouldn't need the `proc` workaround here to act as a closure around the original logger state:

```suggestion
  def initialize(host, port = 80, context = {}, ssl = nil, ssl_version = nil, proxies = nil, username = '', password = '', comm: nil, subscriber: nil)
```